test/evp_test.c

```diff
@@ -1622,7 +1622,8 @@ static int mac_test_run_mac(EVP_TEST *t)
         t->err = "MAC_CREATE_ERROR";
         goto err;
     }
-    size_before_init = EVP_MAC_CTX_get_mac_size(ctx);
+    if (fips_provider_version_gt(libctx, 3, 2, 0))
```

Comment:
this is likely just more of a question: is the EVP_MAC code codified in the fips provider?  i.e. if we use an older fips provider with a newer libcrypto, is the fix for EVP_MAC picked up?

Reply:
No, all crypto (for FIPS approved algorithms) must be inside the FIPS provider. So no, if you use older fips provider with new libcrypto, this bug will still be there, that is the reason we need this condition to actually apply the check.